TypeSystem/flow-analysis/promotion_switch_A05_t01.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion For now (May, 2025) promotion in switch statement and
+/// expressions are not specified yet.
+/// TODO (sgrekhov): update when specified
+///
+/// @description Checks that pattern match doesn't promote the variable when a
+/// case scope is shared abd cases have different types.
+/// @author sgrekhov22@gmail.com
+
+import '../../Utils/static_type_helper.dart';
+
+test1(Object? x) {
+  switch (x) {
+    case int v:
+    case num _:
```

Comment:
I'd think there should be two control flow paths to line 19: Matching the pattern in line 17, or matching the pattern in line 18. At the join, the promotion chains should be combined; `joinV` does this by taking the intersection of the two promotion chains (that is, include exactly the types that occur in both promotion chains, preserving the order), so that would be the empty list here.

OK, so it's not because the cases have different types, it's because these promotion chains happen to have the empty intersection.

So it would be interesting to have a case where the intersection is non-empty (like `[Object, num]` and `[Object, int]`, or `[Object, int]` and `[num, int]`).

Reply:
Perhaps the logic doesn't actually treat this situation as a full-fledged join, but in that case it would be nice to know it. ;)